lib/kdl/include/kdl/vector_utils.h

```diff
@@ -190,6 +190,84 @@ namespace kdl {
         return result;
     }
 
+    /**
+     * Returns a slice of the given vector starting at offset and with count elements.
+     *
+     * If the given offset is not less than the number of elements of v, then an empty vector is returned. The returned
+     * vector contains at most count elements from the given vector. If the given count is too large, i.e. it indicates
+     * to include elements beyond the end of the given vector, then count is adjusted accordingly.
+     *
+     * The elements are copied into the returned vector.
+     *
+     * @tparam T the element type
+     * @tparam A the allocator type
+     * @param v the vector to return a slice of
+     * @param offset the offset of the first element to return
+     * @param count the number of elements to return
+     * @return a vector containing the slice of the given vector
+     */
+    template <typename T, typename A>
+    std::vector<T, A> vec_slice(const std::vector<T, A>& v, const std::size_t offset, std::size_t count) {
```

Comment:
I agree that this is inconsistent, but I'd rather resolve it by making out-of-bounds-access undefined as in most of the other functions. `vec_at` is a special case because it follows the std api, where `at` is also protected against out-of-bounds access.